mmdet/core/hook/wandblogger_hook.py

```diff
@@ -0,0 +1,455 @@
+        logs the predicted bounding boxes along with the ground truth at every
+        evaluation interval. This depends on the `EvalHook` whose priority is
+        more than `MMDetWandbHook`. Also note that the data is just logged once
+        and subsequent evaluation tables uses reference to the logged data
+        to save memory usage. Please refer to
+        https://docs.wandb.ai/guides/data-vis to learn more about W&B Tables.
+
+    For more details check out W&B's MMDetection docs:
+    https://docs.wandb.ai/guides/integrations/mmdetection
+
+    ```
+    Example:
+        log_config = dict(
+            interval=10,
+            hooks=[
+                dict(type='MMDetWandbHook',
+                     init_kwargs={
+                         'entity': WANDB_ENTITY,
+                         'project': WANDB_PROJECT_NAME
+                     },
+                     interval=10,
+                     log_checkpoint=True,
+                     log_checkpoint_metadata=True,
+                     num_eval_images=100)
+            ])
+    ```
+
+    Args:
+        init_kwargs (dict): A dict passed to wandb.init to initialize
+            a W&B run. Please refer to https://docs.wandb.ai/ref/python/init
+            for possible key-value pairs.
+        interval (int): Logging interval (every k iterations).
+            Default 10.
+        log_checkpoint (bool): Save the checkpoint at every checkpoint interval
+            as W&B Artifacts. Use this for model versioning where each version
+            is a checkpoint.
+            Default: False
+        log_checkpoint_metadata (bool): Log the evaluation metrics computed
+            on the validation data with the checkpoint, along with current
+            epoch as a metadata to that checkpoint.
+            Default: True
+        num_eval_images (int): Number of validation images to be logged.
+            Default: 100
+    """
+
+    def __init__(self,
+                 init_kwargs=None,
+                 interval=10,
+                 log_checkpoint=False,
+                 log_checkpoint_metadata=False,
+                 num_eval_images=100,
+                 **kwargs):
+        super(MMDetWandbHook, self).__init__(init_kwargs, interval, **kwargs)
+
+        self.log_checkpoint = log_checkpoint
+        self.log_checkpoint_metadata = log_checkpoint_metadata
+        self.num_eval_images = num_eval_images
+        self.log_evaluation = True
+        self.log_eval_metrics = True
+        self.best_score = 0
+        self.val_step = 0
+
+    @master_only
+    def before_run(self, runner):
+        super(MMDetWandbHook, self).before_run(runner)
+        self.cfg = self.wandb.config
+        # Check if configuration is passed to wandb.
+        if len(dict(self.cfg)) == 0:
+            warnings.warn(
+                'To log mmdetection Config, '
+                'pass it to init_kwargs of MMDetWandbHook.', UserWarning)
```

Comment:
After consideration. It's no good to modify the inputs of runner. I consider if we can obtain the config in the work_dir.

Reply:
That's a great idea. Let me try to grab the config from the `work_dir`.

Reply:
We have saved the configuration so you can read it directly.

Reply:
Hey @hhaAndroid can you point me to a code where the configuration is saved. I couldn't find any saved configuration in the `work_dir`.

Reply:
Although it is stored, but it is not easy to read, I am trying to figure out a way

Reply:
```python
dict(type='MMDetWandbHook',
        init_kwargs={
                'project': 'atss_r50_fpn_1x_coco'
        },
       interval=10,
        log_checkpoint=True,
        log_checkpoint_metadata=True,
        num_eval_images=100,
        saved_config_name='{{ fileBasename }}')


# Load config.json file from work_dir
load_config_path = osp.join(runner.work_dir, self.saved_config_name)
if osp.exists(load_config_path):
      self.wandb.save(load_config_path)
else:
     warnings.warn('`saved_config_name` does not exist, '
              'skip saving the config file directly', UserWarning)
```

Reply:
Hey I am not clear about `args.config`. 

Also when I use '{{ fileBasename }}' gives `/home/ayut/MMDet/mmdetection/work_dirs/faster_rcnn_r50_fpn_1x_coco/default_runtime.py` which I feel is just a small fraction of the complete configuration. 

The current solution that's in the PR is capable of logging the complete config.

Reply:
Hey @hhaAndroid, apologies to unresolve this conversation but just wanted to be sure if my method to log config is acceptable?

I was not able to make your solution work as explained above.